### PR TITLE
Start tokenizing and parsing whitespace

### PR DESCRIPTION
**Pull request by sydneyli**
Should leave all functionality the same. PR maintains all prior functionality (parser understands, but throws away whitespace information). Next PR will have the parser be able to retain whitespace information conditionally!

Now that we're consuming whitespace during lexing rather than parsing, the nostrip flag must be implemented at the lexing layer. Commit [`4f15118`](https://github.com/etingof/apacheconfig/commit/4f15118fb81a723e552a745043cd78d3c1e94855) is my first shot at this, but there may be a more canonical way to construct different lexing rules via PLY.

### CODE REVIEW

File: apacheconfig/lexer.py [hunk 1]
```diff
@@ -224,16 +227,18 @@ def t_multiline_OPTION_AND_VALUE(self, t):
         t.lexer.begin('INITIAL')
 
         value = t.lexer.lexdata[t.lexer.code_start:t.lexer.lexpos + 1]
+        value = self._remove_trailing_whitespace(value)
+        t.lexer.lexpos = t.lexer.code_start + len(value)
         t.lexer.lineno += len(re.findall(r'\r\n|\n|\r', value))
-        value = value.replace('\\\n', '').replace('\r', '').replace('\n', '')
+        value = re.sub(r'(\\\n|\r|\n)', '', value)
```
Comment (etingof) [inline apacheconfig/lexer.py]: Oh, yes!

File: apacheconfig/parser.py [hunk 2]
```diff
@@ -74,50 +74,93 @@ def parse(self, text):
...
+        """
+        if len(p) == 3:
+            p[0] = p[1:][1]
+        else:
+            p[0] = p[1:][0]
+
     def p_contents(self, p):
-        """contents : contents item
-                    | item
+        """contents : contents miditem
```
Comment (etingof) [inline apacheconfig/parser.py]: nit: It feels like this BNF can be slightly reduced.

File: tests/unit/test_parser.py [hunk 3]
```diff
@@ -23,13 +23,13 @@ def testOptionAndValue(self):
 
         parser = ApacheConfigParser(ApacheConfigLexer(), start='statement')
 
-        ast = parser.parse('a b\n')
```
Comment (etingof) [inline tests/unit/test_parser.py]: Should we also have the original test case with a trailing new line as well?

File: apacheconfig/parser.py (hunk 2 above)
Comment (sydneyli) [reply apacheconfig/parser.py]: Any suggestions? Also happy to explain some of the choices here in the comments, like splitting startitem/miditem!

Comment (etingof) [reply apacheconfig/parser.py]: That's a nitpick anyway, let's push this into a TODO and get back to this later?

Comment (sydneyli) [reply apacheconfig/parser.py]: Sounds good! Tests are passing now-- let's merge this in?

**Comment by etingof** [comment]
Awesome, thank you!

**Pull request merged**